### PR TITLE
add support for additional functions and sortAscending

### PR DESCRIPTION
**Pull request by MintyOrb**
for sortAscending, getFilteredItemElements, and getElementItems. Also remove whitespace. I didn't run the gulp tasks...

### CODE REVIEW

**Comment by David-Desmaisons** [comment]
Thanks for your PR. I will analyze it and get back to you.

File: src/vue_isotope.js [hunk 1]
```diff
@@ -91,20 +91,22 @@
           return;
 
         this.listen()
-         
-        this.iso.remove(removed)
-        this.iso.insert(added)
-        this.iso._requestUpdate()      
+
```
Comment (David-Desmaisons) [inline src/vue_isotope.js:94]: I am not sure why you need to check for `this.iso`. Did you find edge cases where it was null?

File: src/vue_isotope.js [hunk 2]
```diff
@@ -115,18 +117,18 @@
...
                 this.iso._requestUpdate();
               });
-            });  
+            });
           }).flatten().value();
         },
 
         sort (name) {
-          this.arrange({sortBy  :name})
+          this.arrange({sortBy: name, sortAscending: this.options.sortAscending})
```
Comment (David-Desmaisons) [inline src/vue_isotope.js:125]: Did you try to see waht happen if sortAscending is undefined?

**Review by David-Desmaisons — COMMENTED** [review]
Hello @bornytm , dis you take a look at my comments? thanks

**Comment by MintyOrb** [comment]
Ah, no sorry- I didn't see your comments until now.

File: src/vue_isotope.js (hunk 1 above)
Comment (MintyOrb) [reply src/vue_isotope.js:94]: I found that checking for this.iso was necessary when using vue-router and changing routes. Without it I get the error `TypeError: Cannot read property 'remove' of undefined` upon changing to the page with the isotope instances and they no longer function when trying to add/remove elements. I don't understand why, but after getting this error I can longer change routes `<router-link :to="/whatever"><router-link>` either. I seems to be related to having multiple vue-isotope instances.

File: src/vue_isotope.js (hunk 2 above)
Comment (MintyOrb) [reply src/vue_isotope.js:125]: Leaving sortAscending undefined still allows for sorting and changing which parameter to sort on (ex. rating vs length) but it was impossible to change the sort order from ascending to descending from within the vue componant data (from highest rating first to lowest first for example). Setting `  sortAscending: this.sortAscending,` to true/false has no effect.

**Comment by David-Desmaisons** [comment]
@bornytm , I am merging your PR. I think I will perform some minor tweaks on the improvement you provided and publish a new version

**Comment by David-Desmaisons** [comment]
@bornytm , I changed the sort API: it continues to work with a string but you can pass also an object with sortAscending setted if you want. This gives the functionality that was missing and remains coherent with the layout API.

**Comment by megahero** [comment]
The current master no longer has the fix for toggling the sortAscending dynamically by using a variable. Has the method change for the sortAscending at any time?

**Pull request merged**